### ISSUE TITLE
Compile issues on 32-bit Linux platforms (with -Werror)

### ISSUE THREAD

**Issue by garybuhrmaster** [Yubico/libfido2#210]
**What version of libfido2 are you using?**

libfido 1.5.0

**What operating system are you running?**

Fedora Linux

**What application are you using in conjunction with libfido2?**

N/A

**How does the problem manifest itself?**

Building application

**Is the problem reproducible?**

Yes

**What are the steps that lead to the problem?**

Attempt to build libfido2 on a 32-bit platform (i686, armv7hl).  While this was a Fedora build I suspect the same issues will occur on other 32-bit platforms.

During compile, the following warnings (which turn into errors for the package build) are produced.

As I recall 64-bit timespec support on 32-bit platforms is not available before glibc 2.32 and requires the appropriate changes in the application to using the new syscall.

While I suspect fewer would want/need 32-bit x86 support (32-bit x86 support is in various stages of being dropped by many distributions), 32-bit arm support is important.

```

/builddir/build/BUILD/libfido2-1.5.0/src/hid_linux.c: In function 'timespec_to_ms':
/builddir/build/BUILD/libfido2-1.5.0/src/hid_linux.c:388:35: error: comparison is always false due to limited range of data type [-Werror=type-limits]
  388 |  if (ts->tv_sec < 0 || ts->tv_sec > INT64_MAX / 1000LL ||
      |                                   ^
/builddir/build/BUILD/libfido2-1.5.0/src/hid_linux.c:389:49: error: comparison is always false due to limited range of data type [-Werror=type-limits]
  389 |      ts->tv_nsec < 0 || ts->tv_nsec / 1000000LL > INT64_MAX)
      |                                                 ^

```

**Comment by martelletto** [comment]
Hi,

Thank you for the comprehensive report. The easiest solution is to disable `-Wtype-limits` by exporting `CFLAGS=-Wno-type-limits`. Going forward, we might add a cast or include `-Wno-type-limits` in the list of default build options for 32-bit platforms, pending further consideration.

-p.

**Comment by martelletto** [comment]
Fixed in 7a17a4e. My sincere apologies for the breakage. I will look into adding a 32-bit CI job (x86 or ARM) when I find the time. Thank you for the detailed report.

**Comment by garybuhrmaster** [comment]
No apologies necessary, anyone doing cross platform development has gotten bitten by type issues at some point.

Thanks for the quick fix.
